### PR TITLE
many-ports: Update supports field

### LINKED ISSUE #20029: [geotrans] Download from https instead of ftp to speed up
****

**Problem**
Downloading geotrans takes too much time as indicated in its [port file](https://github.com/microsoft/vcpkg/blob/030cfaa24de9ea1bbf0a4d9c615ce7312ba77af1/ports/geotrans/portfile.cmake).

**Proposed solution**
It is possible to download the source package from [a https url](https://earth-info.nga.mil/php/download.php?file=wgs-mastertgz) given at [NGA website](https://earth-info.nga.mil/index.php?action=home). So, the download URL in the port file should be updated.

### PR DESCRIPTION
**Pull request by autoantwort**
For https://github.com/microsoft/vcpkg-tool/pull/184

And fixes #20029

### CODE REVIEW

File: ports/behaviortree-cpp/vcpkg.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "behaviortree-cpp",
   "version": "3.5.6",
+  "port-version": 1,
   "description": "Behavior Trees Library in C++.",
   "homepage": "https://www.behaviortree.dev",
-  "supports": "!uwp & !osx",
```
Comment (PhoebeHui) [inline ports/behaviortree-cpp/vcpkg.json:6]: behaviortree-cpp with uwp cascade in CI testing, so'!uwp' should not be removed?

```
behaviortree-cpp:x64-uwp:    cascade
behaviortree-cpp:arm-uwp:    cascade
```

File: ports/bitserializer/vcpkg.json [hunk 2]
```diff
@@ -1,10 +1,9 @@
 {
   "name": "bitserializer",
   "version": "0.10",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Core part of C++ 17 library for serialization to JSON, XML, YAML",
   "homepage": "https://bitbucket.org/Pavel_Kisliak/bitserializer",
-  "supports": "!(arm | osx)",
```
Comment (PhoebeHui) [inline ports/bitserializer/vcpkg.json:7]: Why remove this? I noticed you have added it to ci.baseline.txt file.

```
bitserializer:x64-osx:       skip
bitserializer:arm64-windows:       skip
bitserializer:arm-uwp:       skip
```

File: ports/libxmlpp/vcpkg.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
   "name": "libxmlpp",
   "version": "5.0.0",
+  "port-version": 1,
   "description": "a C++ wrapper for the libxml XML parser library.",
-  "supports": "!(windows & static)",
```
Comment (PhoebeHui) [inline ports/libxmlpp/vcpkg.json:5]: libxmlpp with windows static triplet cascaded in CI testing, so'!(windows & static)' should not be removed?

```
  libxmlpp:x64-windows-static:    cascade:
  libxmlpp:x64-windows-static-md:    cascade
```

File: ports/xnnpack/vcpkg.json [hunk 4]
```diff
@@ -1,9 +1,10 @@
 {
   "name": "xnnpack",
   "version-date": "2021-05-17",
+  "port-version": 1,
   "description": "High-efficiency floating-point neural network inference operators for mobile, server, and Web",
   "homepage": "https://github.com/google/XNNPACK",
-  "supports": "!arm & !uwp",
+  "supports": "!uwp",
```
Comment (PhoebeHui) [inline ports/xnnpack/vcpkg.json:7]: xnnpack with arm triplet cascaded in CI testing, so'!arm' should not be removed?

```
 xnnpack:arm64-windows:    cascade
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@autoantwort, thanks for your PR!

BTW, I checked all related ports in CI testing, except the following ports, others tested and passed in CI testing.

File: ports/behaviortree-cpp/vcpkg.json (hunk 1 above)
Comment (autoantwort) [reply ports/behaviortree-cpp/vcpkg.json:6]: Ah zeromq does not work on uwp, will add it back

File: ports/bitserializer/vcpkg.json (hunk 2 above)
Comment (autoantwort) [reply ports/bitserializer/vcpkg.json:7]: Tested all tree locally and they build without problems.

File: ports/libxmlpp/vcpkg.json (hunk 3 above)
Comment (autoantwort) [reply ports/libxmlpp/vcpkg.json:5]: Right, glib does not build on windows-static

File: ports/xnnpack/vcpkg.json (hunk 4 above)
Comment (autoantwort) [reply ports/xnnpack/vcpkg.json:7]: Right, cpuinfo does not build

**Review by PhoebeHui — APPROVED** [review]
Thanks for your updates!

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the fixes!

**Pull request merged**